src/main/java/cz/brmlab/yodaqa/flow/dashboard/SourceIDGenerator.java

```diff
@@ -0,0 +1,23 @@
+package cz.brmlab.yodaqa.flow.dashboard;
+
+/**
+ * Created by nhl on 7/9/15.
+ */
+
+
+/**
+ * This class should generate unique ID's for sources!
+ * The ID's are unique for the session
+ * It is thread-safe
+ */
+public final class SourceIDGenerator {
+	/*Singleton*/
+	volatile int IDCounter = 0;
```

Comment:
https://stackoverflow.com/questions/9665483/why-using-volatile-with-synchronized-block

Probably is redundant for integers, though